### PR TITLE
Release page: Time filter

### PR DESCRIPTION
**Pull request by hedgehog1833**
Time filter added on releases page for all upcoming, next month and last month. I also renamed "Earliest first" and "lates first" to "Ascending date" and "Descending date" as it confuses me every time I use this filter (and I also got this as feedback already^^)

### CODE REVIEW

**Comment by hedgehog1833** [comment]
[Releases page: Time Filter](https://trello.com/c/CZb1kwRU/231-releases-page-time-filter)

File: webapp/src/main/resources/static/ts/src/service/releases-service.ts [hunk 1]
```diff
@@ -69,6 +84,10 @@ export class ReleasesService {
         this.sortDescRb.checked = directionParamValue === ReleasesService.SORT_DIRECTION_DESC_PARAM_VALUE;
         this.sortAscRb.checked = !this.sortDescRb.checked;
         this.searchField.value = searchQueryParamValue;
+        this.timeAllUpcomingRb.checked = dateToParamValue === "";
+        this.timeNextMonthRb.checked =
+            !this.timeAllUpcomingRb.checked && Date.parse(dateFromParamValue) == Date.parse(this.dateService.today());
```
Comment (DanielW1987) [inline webapp/src/main/resources/static/ts/src/service/releases-service.ts:89]: ...to be sure

```suggestion
            !this.timeAllUpcomingRb.checked && Date.parse(dateFromParamValue) === Date.parse(this.dateService.today());
```

File: webapp/src/main/resources/templates/frontend/releases.html [hunk 2]
```diff
@@ -113,35 +113,23 @@ <h1 class="h4">Releases</h1>
...
+                                <label class="custom-control-label" for="time-all-upcoming-rb">All upcoming</label>
+                            </div>
+                            <div class="custom-control custom-radio">
+                                <input type="radio" id="time-next-month-rb" name="time" class="custom-control-input">
+                                <label class="custom-control-label" for="time-next-month-rb">Next month</label>
+                            </div>
+                            <div class="custom-control custom-radio">
+                                <input type="radio" id="time-last-month-rb" name="time" class="custom-control-input">
+                                <label class="custom-control-label" for="time-last-month-rb">Last month</label>
+                            </div>
```
Comment (DanielW1987) [inline webapp/src/main/resources/templates/frontend/releases.html:130]: I don't know if the terms "next month" and "last month" might be a bit confusing. Actually, it's the next 30 days or the last 30 days. "Next month" refers more to a point in time than a period of time. What do you think.

Yes, that was already wrong in my template ;)

File: webapp/src/main/resources/templates/frontend/releases.html [hunk 3]
```diff
@@ -89,11 +89,11 @@ <h1 class="h4">Releases</h1>
...
                                 <div class="custom-control custom-radio">
                                     <input type="radio" id="sort-asc-rb" name="releases" class="custom-control-input" checked="">
-                                    <label class="custom-control-label" for="sort-asc-rb">Earliest first</label>
+                                    <label class="custom-control-label" for="sort-asc-rb">Ascending date</label>
                                 </div>
                                 <div class="custom-control custom-radio">
                                     <input type="radio" id="sort-desc-rb" name="releases" class="custom-control-input">
-                                    <label class="custom-control-label" for="sort-desc-rb">Latest first</label>
+                                    <label class="custom-control-label" for="sort-desc-rb">Descending date</label>
                                 </div>
```
Comment (DanielW1987) [inline webapp/src/main/resources/templates/frontend/releases.html:97]: Now that you mention it...was stupid before.

File: webapp/src/main/resources/templates/frontend/releases.html [hunk 4]
```diff
@@ -89,11 +89,11 @@ <h1 class="h4">Releases</h1>
                             <form id="sorting-form">
                                 <div class="custom-control custom-radio">
                                     <input type="radio" id="sort-asc-rb" name="releases" class="custom-control-input" checked="">
-                                    <label class="custom-control-label" for="sort-asc-rb">Earliest first</label>
+                                    <label class="custom-control-label" for="sort-asc-rb">Ascending date</label>
```
Comment (DanielW1987) [inline webapp/src/main/resources/templates/frontend/releases.html:92]:
```suggestion
                                    <label class="custom-control-label" for="sort-asc-rb">Ascending</label>
```

File: webapp/src/main/resources/templates/frontend/releases.html [hunk 5]
```diff
@@ -89,11 +89,11 @@ <h1 class="h4">Releases</h1>
...
                                 </div>
                                 <div class="custom-control custom-radio">
                                     <input type="radio" id="sort-desc-rb" name="releases" class="custom-control-input">
-                                    <label class="custom-control-label" for="sort-desc-rb">Latest first</label>
+                                    <label class="custom-control-label" for="sort-desc-rb">Descending date</label>
```
Comment (DanielW1987) [inline webapp/src/main/resources/templates/frontend/releases.html:96]:
```suggestion
                                    <label class="custom-control-label" for="sort-desc-rb">Descending</label>
```

File: webapp/src/main/resources/static/ts/src/clients/releases-rest-client.ts [hunk 6]
```diff
@@ -28,10 +28,13 @@ export class ReleasesRestClient {
         const sort = this.urlService.getParameterFromUrl("sort");
         const direction = this.urlService.getParameterFromUrl("direction");
         const query = this.urlService.getParameterFromUrl("query");
+        const dateFrom = this.urlService.getParameterFromUrl("dateFrom");
+        const dateTo = this.urlService.getParameterFromUrl("dateTo");
         axiosConfig.params = {
             page: this.urlService.getPageFromUrl(),
             size: 30,
-            dateFrom: this.dateService.today(),
+            dateFrom: dateFrom === "" ? this.dateService.today() : dateFrom,
```
Comment (DanielW1987) [inline webapp/src/main/resources/static/ts/src/clients/releases-rest-client.ts:36]: can shorten with

```suggestion
            dateFrom: dateFrom || this.dateService.today(),
```

File: webapp/src/main/resources/templates/frontend/releases.html (hunk 2 above)
Comment (hedgehog1833) [reply webapp/src/main/resources/templates/frontend/releases.html:130]: Yeeeah the 30 days version sounds better, I changed it.

**Review by DanielW1987 — APPROVED** [review]

**Pull request merged**